src/ssl.c

```diff
@@ -17989,14 +17995,17 @@ size_t wolfSSL_get_client_random(const WOLFSSL* ssl, unsigned char* out,
         if (ssl->hsHashes)
             (void)InitHandshakeHashes(ssl);
 
-#ifdef SESSION_CERTS
-        ssl->session->chain.count = 0;
-#endif
 #ifdef KEEP_PEER_CERT
         FreeX509(&ssl->peerCert);
         InitX509(&ssl->peerCert, 0, ssl->heap);
 #endif
 
+        wolfSSL_SESSION_free(ssl->session);
```

Comment:
Good fix!

Reply:
Actually this should only be done if the session is invalid ie: the handshake didn't complete.